bin/check_dockerfiles.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+BASE_DIR="$(cd "$(dirname "${0}")" && pwd -P)"
+ISTIO_ROOT="$(cd "$(dirname "${BASE_DIR}")" && pwd -P)"
+CD_TMPFILE=$(mktemp /tmp/check_dockerfile.XXXXXX)
+HL_TMPFILE=$(mktemp /tmp/hadolint.XXXXXX)
+
```

Comment:
please submit this binary to the repo istio/pkg first.  Once it passes review there link the review here and we can approve.  FWIW, I don't see any problems with the bash, but I'm not real clear we want to run docker run -rm -i within kind.  I'd prefer someone that understands our new Kind CI approach review the docker run.  That would happen in istio/pkg repo.